### ISSUE TITLE
Unable to reproduce claimed 90-93% RTE accuracy with GPT-Neo-1.3B

### ISSUE THREAD

**Issue by timstoop** [Anima-Core/an1-core#2]
## Issue

Following the README instructions exactly:

```bash
python experiments/reproduce_rte.py \
  --model-name EleutherAI/gpt-neo-1.3B \
  --max-train-examples 2490 \
  --max-val-examples 277 \
  --epochs 10
```

**Expected:** 90-93% accuracy (per README line 29)

**Actual:** ~60% accuracy

## Testing details

Tested across multiple model sizes:
- GPT-Neo-125M (full precision): ~57-60% accuracy
- GPT-Neo-1.3B (8-bit quantization): ~60% accuracy  
- GPT-Neo-2.7B (8-bit quantization): ~62% accuracy

All results consistently show ~60% range, significantly below the claimed 90-93%.

## Questions

1. Are there missing hyperparameters or configuration needed to achieve the claimed results?
2. Does the 90-93% claim require proprietary components (FPU/AN4) that are explicitly excluded from this reference implementation?
3. If proprietary components are required, should the README clarify expected results without those components?

The README states this is a "scientific reference implementation" but the 30-point accuracy gap suggests either missing methodology or that the claimed results are not achievable with the provided code.

## Environment

- Hardware: RTX 3050 8GB
- PyTorch 2.1.0
- transformers 4.45.0
- Quantization: 8-bit for 1.3B/2.7B models (125M tested in full precision)

Note: 8-bit quantization typically causes <2% accuracy degradation, so this cannot explain the 30-point gap.
